test/powershell/Modules/Microsoft.PowerShell.Management/Restart-Computer.Tests.ps1

```diff
@@ -0,0 +1,100 @@
+# the testhook for restart-computer is the same as for stop-computer
+$restartTesthookName = "TestStopComputer"
+$restartTesthookResultName = "TestStopComputerResults"
+$DefaultResultValue = 0
+
+try 
+{
+    # set up for testing
+    $PSDefaultParameterValues["it:skip"] = ! $IsWindows
+    Enable-Testhook -testhookName $restartTesthookName
+
+    Describe "Restart-Computer" -Tag Feature {
+        # if we throw in BeforeEach, the test will fail and the restart will not be called
+        BeforeEach {
+            if ( ! (Test-TesthookIsSet -testhookName $restartTesthookName) ) {
+                throw "Testhook '${restartTesthookName}' is not set"
+            }
+        }
+
+        AfterEach {
+            Set-TesthookResult -testhookName $restartTesthookResultName -value $defaultResultValue
+        }
+
+        It "Should restart the local computer" {
+            Set-TesthookResult -testhookName $restartTesthookResultName -value $defaultResultValue
+            Restart-Computer -ErrorAction Stop| Should BeNullOrEmpty
+        }
+
+        It "Should support -computer parameter" {
+            Set-TesthookResult -testhookName $restartTesthookResultName -value $defaultResultValue
+            $computerNames = "localhost",(hostname)
+            Restart-Computer -Computer $computerNames -ErrorAction Stop| Should BeNullOrEmpty
+        }
+
+        It "Should support WsmanAuthentication types" {
+            $authChoices = "Default","Basic","Negotiate","CredSSP","Digest","Kerberos"
+            foreach ( $auth in $authChoices ) {
+                Restart-Computer -WsmanAuthentication $auth | Should BeNullOrEmpty

```

Comment:
since this is a straight shot into the cim session creation, there isn't a lot to be gained here, but for coverage of the parameter setting I added this test. There doesn't seem to be any property on the cimsession which describes how the session was created. I could pull all but one which would provide the coverage.